static/app/views/detectors/components/forms/cron/detect.tsx

```diff
@@ -0,0 +1,282 @@
+import {FieldWrapper} from 'sentry/components/forms/fieldGroup/fieldWrapper';
+import NumberField from 'sentry/components/forms/fields/numberField';
+import SelectField from 'sentry/components/forms/fields/selectField';
+import TextField from 'sentry/components/forms/fields/textField';
+import {Container} from 'sentry/components/workflowEngine/ui/container';
+import Section, {SectionSubHeading} from 'sentry/components/workflowEngine/ui/section';
+import {timezoneOptions} from 'sentry/data/timezones';
+import {t, tct, tn} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {SelectValue} from 'sentry/types/core';
+import {useCronDetectorFormField} from 'sentry/views/detectors/components/forms/cron/fields';
+import {ScheduleType} from 'sentry/views/insights/crons/types';
+import {getScheduleIntervals} from 'sentry/views/insights/crons/utils';
+import {crontabAsText} from 'sentry/views/insights/crons/utils/crontabAsText';
+
+const SCHEDULE_OPTIONS: Array<SelectValue<string>> = [
+  {value: ScheduleType.CRONTAB, label: t('Crontab')},
+  {value: ScheduleType.INTERVAL, label: t('Interval')},
+];
+
+const DEFAULT_CRONTAB = '0 0 * * *';
+
+// In minutes
+const DEFAULT_MAX_RUNTIME = 30;
+const DEFAULT_CHECKIN_MARGIN = 1;
+const CHECKIN_MARGIN_MINIMUM = 1;
+const TIMEOUT_MINIMUM = 1;
+
+type Props = {
+  isEditing: boolean;
+};
+
+function ScheduleTypeField() {
+  return (
+    <Fragment>
+      <SelectField
+        name="scheduleType"
+        label={t('Schedule Type')}
+        hideLabel
+        options={SCHEDULE_OPTIONS}
+        defaultValue={ScheduleType.CRONTAB}
+        orientInline
+        required
+        stacked
+        inline={false}
+      />
+    </Fragment>
+  );
+}
+
+function Schedule() {
+  const theme = useTheme();
+  const scheduleCrontab = useCronDetectorFormField('scheduleCrontab');
+  const scheduleIntervalValue = useCronDetectorFormField('scheduleIntervalValue');
+  const scheduleType = useCronDetectorFormField('scheduleType');
+
+  const parsedSchedule =
+    scheduleType === 'crontab' ? crontabAsText(scheduleCrontab) : null;
+
+  if (scheduleType === 'crontab') {
+    return (
+      <InputGroup removeFieldPadding>
+        <ScheduleTypeField />
+        <MultiColumnInput columns="1fr 2fr">
+          <TextField
+            name="scheduleCrontab"
+            label={t('Crontab Schedule')}
+            hideLabel
+            placeholder="* * * * *"
+            defaultValue={DEFAULT_CRONTAB}
+            css={css`
+              input {
+                font-family: ${theme.text.familyMono};
+              }
+            `}
+            required
+            stacked
+            inline={false}
+          />
+          <SelectField
+            name="timezone"
+            label={t('Timezone')}
+            hideLabel
+            defaultValue="UTC"
+            options={timezoneOptions}
+            required
+            stacked
+            inline={false}
+          />
```

Comment:
i think you'll need to use `preserveOnUnmount` if you want to use `useCronDetectorFormField` on these fields. Otherwise the observable is removed when they are unmounted. You can reproduce this by switching back and forth and editing the crontab. The scheduleCrontab stops updating
```suggestion
          <SelectField
            name="timezone"
            label={t('Timezone')}
            hideLabel
            defaultValue="UTC"
            options={timezoneOptions}
            required
            stacked
            inline={false}
            preserveOnUnmount
          />
```

Reply:
in this case you could also use local state to copy it to on change since the data isn't traveling very far

Reply:
Okay yeah it works much better with preserveOnUnmount!